library/bigip_vlan.py

```diff
@@ -154,300 +152,301 @@
-            return changed
+                err = 'The specified interface "%s" was not found' % ifc
+                raise F5ModuleError(err)
+
+        if key == 'tagged_interfaces':
+            tagged = [dict(name=x, tagged=True) for x in pinterfaces]
+            if self.have is not None:
+                if want != have:
+                    return tagged
+            else:
+                return tagged
 
-        r = self.api.tm.net.vlans.vlan.load(
-            name=name,
-            partition=partition
-        )
-        r.update(**params)
-        r.refresh()
+        if key == 'untagged_interfaces':
+            untagged = [dict(name=x, untagged=True) for x in pinterfaces]
+            if self.have is not None:
+                if want != have:
+                    return untagged
+            else:
+                return untagged
 
-        return True
+    def _have_interfaces(self, ifcs):
+        untagged = [str(x.name) for x in ifcs if hasattr(x, 'untagged')]
+        tagged = [str(x.name) for x in ifcs if hasattr(x, 'tagged')]
+        if untagged:
+            self.have.untagged_interfaces = untagged
+        if tagged:
+            self.have.tagged_interfaces = tagged
 
-    def delete(self):
-        params = dict()
-        check_mode = self.params['check_mode']
+    def present(self):
+        if self.exists():
+            return self.update()
+        else:
+            return self.create()
 
-        params['name'] = self.params['name']
-        params['partition'] = self.params['partition']
+    def absent(self):
+        if self.exists():
+            return self.remove()
+        return False
 
-        self.cparams = camel_dict_to_snake_dict(params)
-        if check_mode:
+    def should_update(self):
+        result = self._update_changed_options()
+        if result:
             return True
+        return False
 
-        dc = self.api.tm.net.vlans.vlan.load(**params)
-        dc.delete()
+    def update(self):
+        self.have, ifcs = self.read_current_from_device()
+        if ifcs:
+            self._have_interfaces(ifcs)
+        if not self.should_update():
+            return False
+        if self.client.check_mode:
+            return True
+        self.update_on_device()
+        return True
 
+    def remove(self):
+        if self.client.check_mode:
+            return True
+        self.remove_from_device()
         if self.exists():
-            raise F5ModuleError("Failed to delete the VLAN")
+            raise F5ModuleError("Failed to delete the user")
         return True
 
-    def exists(self):
-        name = self.params['name']
-        partition = self.params['partition']
-        return self.api.tm.net.vlans.vlan.exists(
-            name=name,
-            partition=partition
+    def create(self):
+        self._set_changed_options()
+        if self.client.check_mode:
+            return True
+        self.create_on_device()
+        return True
+
+
+class VlanManager(BaseManager):
```

Comment:
well you wanted to subclass, i have not seen a reason for it either, just going by what we have done last time